### PR TITLE
Nitunit: mass compile non-simple docunits

### PR DESCRIPTION
**Pull request by privat**
When docunits are simple (just statements), a single module is generated for all.
Unfortunately, when docunits contains classes or importations, a specific module must be generated for each one of them.
Compiling all these module takes most of the nitunit elapsed time.

This PR tries to compile these non-simple modules at once, thanks to nitc that features invoking it with multiple programs.

Testing with `nitunit lib/popcorn/README.md`

Before:
* cold ccache: 1m3,832s
* hot ccache: 0m23,063s

After:
* cold ccache: 0m44,517s (-30%)
* hot ccache: 0m17,778s (-23%)

### CODE REVIEW

**Comment by privat** [comment]
test this please

File: src/testing/testing_doc.nit [hunk 1]
```diff
@@ -293,6 +299,44 @@ class NitUnitExecutor
...
+
+		if files.is_empty then return 0
+
+		toolcontext.info("Compile {dus.length} single(s) doc-unit(s) at once", 1)
+
+		# Mass compile them
+		var nitc = toolcontext.find_nitc
+		var opts = new Array[String]
+		if mmodule != null then
+			opts.add "-I {mmodule.filepath.dirname}"
```
Comment (jcbrinfo) [inline src/testing/testing_doc.nit]: Are you sure it works when the module is inside a group? Example: `cd contrib/foo && nitunit foo::bar::baz`

File: src/testing/testing_doc.nit [hunk 2]
```diff
@@ -293,6 +299,44 @@ class NitUnitExecutor
 		var res = toolcontext.safe_exec(cmd)
 		return res
 	end
+
+	# Compile an unit file and return the compiler return code
+	#
+	# Can terminate the program if the compiler is not found
+	private fun compile_single_docunits(dus: Array[DocUnit]): Int
+	do
```
Comment (jcbrinfo) [inline src/testing/testing_doc.nit]: The return value is not documented and never used.

File: src/testing/testing_doc.nit [hunk 3]
```diff
@@ -175,7 +180,7 @@ class NitUnitExecutor
 
 		if res != 0 then
 			# Compilation error.
-			# They will be executed independently
+			# They should be generated and compiler independently
```
Comment (Morriar) [inline src/testing/testing_doc.nit]: compiled*

File: src/testing/testing_doc.nit [hunk 4]
```diff
@@ -293,6 +299,44 @@ class NitUnitExecutor
 		var res = toolcontext.safe_exec(cmd)
 		return res
 	end
+
+	# Compile an unit file and return the compiler return code
```
Comment (Morriar) [inline src/testing/testing_doc.nit]: I think it's `a unit`:

https://owl.english.purdue.edu/owl/resource/591/01/

**Comment by privat** [comment]
test this please

**Comment by privat** [comment]
reroll: fix minor doc and spell issues and update tests sav

**Pull request merged**